### DISCUSSION TITLE
How to provide number of partitions and replication factor in KafkaBroker

### DISCUSSION THREAD

**Discussion by didhat** [ag2ai/faststream#1821] [Q&A] [answered]
Hello, I use confluent kafka broker and it works nice for me. Bud I have problem, I can not set up number of partitions and replication factor for topics and default values  `num_partitions=1, replication_factor=1` are not suitable for my environment. Because of this I have to turn off automatic topic creation and do it manually. I found code code that create topics and there `num_partitions` and `replication_factor` are hardcoded. Is there any a special reason why this done this way or is it possible to change this behavior and add the ability to set these parameters yourself?

https://github.com/airtai/faststream/blob/7c069db7ff28bb43451aaa4dbffd84e8c261567a/faststream/confluent/client.py#L438

**Comment by kumaranvpl** [comment, marked as answer]
Hello @didhat,

That was created as a temporary fix and the idea behind it was that almost all the time users would have existing topics and faststream merely acts as a bridge for produce/subscribe.

Anyway I agree that this is a bug and I have created an issue for it - https://github.com/airtai/faststream/issues/1827.

Thanks for letting us know and sorry for the late reply.
